### ISSUE TITLE
Support for php:// stream wrappers in \PDepend\Engine

### ISSUE THREAD

**Issue by Sweetchuck** [pdepend/pdepend#361]
The problem is that the `\PDepend\Engine::createFileIterator()` iterates over the `$this->files` which contains a file path like this `php://stdin`, but that file path is added to the `$fileIterator` like this:
`$fileIterator->append(new Iterator(new \GlobIterator($file), $this->fileFilter));`,
and later the `\PDepend\Input\Iterator::getFullPath()` is called, which calls the `::getRealPath()` method of the current item, which is an instance of `\SplFileObject`, and it always returns false.

In short this is the problem:
```php
$item = new \SplFileObject('php://stdin');
$path = $item->getRealPath(); // $path === false
```

Original issue: [phpmd/phpmd#542]
